cloud_testing/kotlin_poc/src/main/kotlin/ftl/gc/GcIosTestMatrix.kt

```diff
@@ -0,0 +1,56 @@
+package ftl.gc
+
+import com.google.api.services.testing.Testing
+import com.google.api.services.testing.model.*
+import ftl.config.YamlConfig
+import ftl.util.Utils
+import ftl.util.Utils.fatalError
+import ftl.util.Utils.join
+import java.util.concurrent.TimeUnit
+
+object GcIosTestMatrix {
+
+    fun build(
+            iosDevice: IosDevice,
+            xcTestGcsPath: String,
+            runGcsPath: String,
+            config: YamlConfig): Testing.Projects.TestMatrices.Create {
+
+        val iOSXCTest = IosXcTest()
+                .setTestsZip(FileReference().setGcsPath(xcTestGcsPath))
+
+        val iOSTestSetup = IosTestSetup()
+                .setNetworkProfile(null)
+
+        val testTimeoutSeconds = TimeUnit.MINUTES.toSeconds(config.testTimeoutMinutes)
+        val testSpec = TestSpecification()
+                .setDisablePerformanceMetrics(config.disablePerformanceMetrics)
+                .setDisableVideoRecording(config.disableVideoRecording)
+                .setTestTimeout("${testTimeoutSeconds}s")
+                .setIosTestSetup(iOSTestSetup)
+                .setIosXcTest(iOSXCTest)
+
+        val matrixGcsPath = join(config.rootGcsBucket, runGcsPath, Utils.uniqueObjectName())
+
+        val resultStorage = ResultStorage()
+                .setGoogleCloudStorage(GoogleCloudStorage().setGcsPath(matrixGcsPath))
+
+        val environmentMatrix = EnvironmentMatrix()
+                .setIosDeviceList(
+                        IosDeviceList().setIosDevices(listOf(iosDevice)))
+
+        val testMatrix = TestMatrix()
+                .setTestSpecification(testSpec)
+                .setClientInfo(ClientInfo().setName("Flank"))
+                .setEnvironmentMatrix(environmentMatrix)
+                .setResultStorage(resultStorage)
+
+        try {
+            return GcTesting.get.projects().testMatrices().create(config.projectId, testMatrix)
+        } catch (e: Exception) {
+            fatalError(e)
+        }
+
+        throw RuntimeException("Failed to create test matrix")
```

Comment:
haven't read much of the code but why is this not handled by `fatalError`?

Reply:
The code doesn't compile if this RuntimeException is removed. I had the same thought.
![image](https://user-images.githubusercontent.com/1173057/40241013-e594406a-5a87-11e8-9cdb-4b12d8e77d51.png)